### PR TITLE
8 digits bin values

### PR DESCRIPTION
**Pull request by mohammedDehairy**
8 digits bin values

### CODE REVIEW

**Review by erenbesel — DISMISSED** [review]

File: AdyenCard/Components/Card/CardComponent.swift [hunk 1]
```diff
@@ -174,13 +173,28 @@ public class CardComponent: PresentableComponent,
...
+    private let panThrottler = Throttler(minimumDelay: CardComponent.Constant.secondsThrottlingDelay)
+    private let binThrottler = Throttler(minimumDelay: CardComponent.Constant.secondsThrottlingDelay)
 }
 
 extension CardComponent: CardViewControllerDelegate {
     
-    func didChangeBIN(_ value: String) {
-        self.cardComponentDelegate?.didChangeBIN(String(value.prefix(Constant.publicBinLength)), component: self)
-        binInfoProvider.provide(for: value, supportedTypes: supportedCardTypes) { [weak self] binInfo in
+    func didChange(pan: String) {
```
Comment (descorp) [inline AdyenCard/Components/Card/CardComponent.swift]: Missing access level. Should be `internal` :)

File: AdyenCard/Components/Card/CardComponentDelegate.swift [hunk 2]
```diff
@@ -10,20 +10,24 @@ import UIKit
 /// Delegate for observing user's activity on `CardComponent`.
 public protocol CardComponentDelegate: AnyObject {
 
-    /// Called when shopper enters PAN in `CardComponent`.
-    /// - Parameter value: Up to 6 first digits in entered PAN.
+    /// Called with the first 6 or 8 digits typed by the shopper in the PAN field in `CardComponent`.
```
Comment (descorp) [inline AdyenCard/Components/Card/CardComponentDelegate.swift:13]:
```suggestion
    /// Called with the first 6 or 11 digits typed by the shopper in the PAN field in `CardComponent`.
```

Comment (mohammedDehairy) [reply AdyenCard/Components/Card/CardComponentDelegate.swift:13]: it'll never be 11, BIN is max 8 digits :-)

File: AdyenCard/Components/Card/CardComponent.swift (hunk 1 above)
Comment (mohammedDehairy) [reply AdyenCard/Components/Card/CardComponent.swift]: right :-D

File: AdyenCard/Components/Card/CardComponentDelegate.swift (hunk 2 above)
Comment (descorp) [reply AdyenCard/Components/Card/CardComponentDelegate.swift:13]: `thresholdBINLength = 11` ?

**UPD**
Ok, this is why it was called "privateBinLength" - it is for internal use only

File: AdyenCard/Components/Card/CardViewController.swift [hunk 3]
```diff
@@ -290,18 +292,21 @@ internal class CardViewController: FormViewController {
     }
 
     private func setupViewRelations() {
-        observe(items.numberContainerItem.numberItem.$binValue) { [weak self] in self?.didReceive(bin: $0) }
+        observe(items.numberContainerItem.numberItem.$panValue) { [weak self] in self?.didChange(pan: $0) }
+        observe(items.numberContainerItem.numberItem.$binValue) { [weak self] in self?.didChange(bin: $0) }
 
         items.button.buttonSelectionHandler = { [weak cardDelegate] in
             cardDelegate?.didSelectSubmitButton()
```
Comment (descorp) [inline AdyenCard/Components/Card/CardViewController.swift:299]: In another PR we should rename `cardDelegate` -> `componentDelegate`

**Review by descorp — APPROVED** [review]

**Review by erenbesel — APPROVED** [review]

**Review by nauaros — APPROVED** [review]

**Pull request merged**